src/spec-node/templatesCLI/publish.ts

```diff
@@ -0,0 +1,96 @@
+import path from 'path';
+import * as os from 'os';
+import { Argv } from 'yargs';
+import { LogLevel, mapLogLevel } from '../../spec-utils/log';
+import { rmLocal } from '../../spec-utils/pfs';
+import { getPackageConfig } from '../../spec-utils/product';
+import { createLog } from '../devContainers';
+import { UnpackArgv } from '../devContainersSpecCLI';
+import { publishOptions } from '../collectionCommonUtils/publish';
+import { getCLIHost } from '../../spec-common/cliHost';
+import { loadNativeModule } from '../../spec-common/commonUtils';
+import { PackageCommandInput } from '../collectionCommonUtils/package';
+import { OCICollectionFileName } from '../collectionCommonUtils/packageCommandImpl';
+import { doTemplatesPackageCommand } from './packageCommandImpl';
+import { getRef, OCICollectionRef } from '../../spec-configuration/containerCollectionsOCI';
+import { doPublishCommand, doPublishMetadata } from '../collectionCommonUtils/publishCommandImpl';
+
+const collectionType = 'template';
+
+export function templatesPublishOptions(y: Argv) {
+    return publishOptions(y, 'template');
+}
+
+export type TemplatesPublishArgs = UnpackArgv<ReturnType<typeof templatesPublishOptions>>;
+
+export function templatesPublishHandler(args: TemplatesPublishArgs) {
+    (async () => await templatesPublish(args))().catch(console.error);
+}
+
+async function templatesPublish({
+    'target': targetFolder,
+    'log-level': inputLogLevel,
+    'registry': registry,
+    'namespace': namespace
+}: TemplatesPublishArgs) {
+    const disposables: (() => Promise<unknown> | undefined)[] = [];
+    const dispose = async () => {
+        await Promise.all(disposables.map(d => d()));
+    };
+
+    const pkg = getPackageConfig();
+
+    const cwd = process.cwd();
+    const cliHost = await getCLIHost(cwd, loadNativeModule);
+    const output = createLog({
+        logLevel: mapLogLevel(inputLogLevel),
+        logFormat: 'text',
+        log: (str) => process.stdout.write(str),
+        terminalDimensions: undefined,
+    }, pkg, new Date(), disposables);
+
+    // Package templates
+    const outputDir = path.join(os.tmpdir(), '/templates-output');
```

Comment:
Make this a unique folder name. Appending `Date.now()` might be enough, but I have started using the `uuid` package in vscode-remote-containers to be on the safe side, e.g.: `/templates-output-${uuid.v4()}`.

Could you apply the same fix in the features code?

Reply:
Sure, added changes. 👍